### PR TITLE
docs: update "Metrics" Plugins

### PR DESCRIPTION
**Pull request by navendu-pottekkat**
Signed-off-by: Navendu Pottekkat <navendupottekkat@gmail.com>

### Description

Updates the documentation of the "Observability/Metrics" Plugins.

Child PR of #6734

### CODE REVIEW

**Comment by navendu-pottekkat** [comment]
// @hf400159 @kwanhur @avinal @yzeng25 @juzhiyuan Please review when you are available.

File: docs/en/latest/plugins/datadog.md [hunk 1]
```diff
@@ -23,62 +29,93 @@ title: datadog
 
 ## Description
 
-`datadog` is a monitoring plugin built into Apache APISIX for seamless integration with [Datadog](https://www.datadoghq.com/), one of the most used monitoring and observability platform for cloud applications. If enabled, this plugin supports multiple powerful types of metrics capture for every request and response cycle that essentially reflects the behaviour and health of the system.
+The `datadog` monitoring Plugin if for seamless integration of APISIX with [Datadog](https://www.datadoghq.com/), one of the most used monitoring and observability platform for cloud applications.
```
Comment (avinal) [inline docs/en/latest/plugins/datadog.md:32]:
```suggestion
The `datadog` monitoring Plugin is for seamless integration of APISIX with [Datadog](https://www.datadoghq.com/), one of the most used monitoring and observability platform for cloud applications.
```

File: docs/en/latest/plugins/datadog.md [hunk 2]
```diff
@@ -23,62 +29,93 @@ title: datadog
...
-To learn more about DogStatsD, please visit [DogStatsD](https://docs.datadoghq.com/developers/dogstatsd/?tab=hostagent) documentation.
+When enabled, the Plugin supports multiple metric capture types for request and response cycles.
 
-This plugin provides the ability to push metrics as a batch to the external Datadog agent, reusing the same datagram socket. In case if you did not receive the log data, don't worry give it some time. It will automatically send the logs after the timer function expires in our Batch Processor.
+This Plugin, pushes its custom metrics to the [DogStatsD](https://docs.datadoghq.com/developers/dogstatsd/?tab=hostagent) server over UDP protocol and comes bundled with [Datadog agent](https://docs.datadoghq.com/agent/).
 
-For more info on Batch-Processor in Apache APISIX please refer.
-[Batch-Processor](../batch-processor.md)
+DogStatsD is basically an implementation of the StatsD protocol which collects the custom metrics for Apache APISIX agent, aggregates it into a single data point, and sends it to the configured Datadog server.
```
Comment (avinal) [inline docs/en/latest/plugins/datadog.md:38]:
```suggestion
DogStatsD implements the StatsD protocol which collects the custom metrics for the Apache APISIX agent, aggregates them into a single data point, and sends it to the configured Datadog server.
```

File: docs/en/latest/plugins/datadog.md [hunk 3]
```diff
@@ -23,62 +29,93 @@ title: datadog
...
-To know more about how to effectively write tags, please visit [here](https://docs.datadoghq.com/getting_started/tagging/#defining-tags)
+```shell
+curl http://127.0.0.1:9080/apisix/admin/plugin_metadata/datadog -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '{}'
+```
 
-## Exported Metrics
+## Exported metrics
 
-Apache APISIX agent, for every request response cycle, export the following metrics to DogStatsD server if the datadog plugin is enabled:
+When the `datadog` Plugin is enabled, APISIX agent exports the following metrics to the DogStatsD server for each request/response cycle:
```
Comment (avinal) [inline docs/en/latest/plugins/datadog.md:90]:
```suggestion
When the `datadog` Plugin is enabled, the APISIX agent exports the following metrics to the DogStatsD server for each request/response cycle:
```

File: docs/en/latest/plugins/datadog.md [hunk 4]
```diff
@@ -23,62 +29,93 @@ title: datadog
...
 
 The metrics will be sent to the DogStatsD agent with the following tags:
 
-> If there is no suitable value for any particular tag, the tag will simply be omitted.
+- `route_name`: Name specified in the Route schema definition. If not present or if the attribute `prefer_name` is set to false, falls back to the Route ID.
+- `service_name`: If a Route has been created with an abstracted Service, the Service name/ID based on the attribute `prefer_name`.
+- `consumer`: If the Route is linked to a Consumer, the username will be added as a tag.
+- `balancer_ip`: IP address of the Upstream balancer that processed the current request.
+- `response_status`: HTTP response status code.
+- `scheme`: Request scheme such as HTTP, gRPC, and gRPCs.
```
Comment (avinal) [inline docs/en/latest/plugins/datadog.md:108]: I guess it would be nice to put these into a tabular form. What do you think?

**Comment by avinal** [comment]
IIRC we agreed upon using `bash` as code highlighting syntax instead of `shell`. Please see https://github.com/apache/apisix/pull/7030#discussion_r871813072. Rest seems good to me

File: docs/en/latest/plugins/datadog.md (hunk 4 above)
Comment (navendu-pottekkat) [reply docs/en/latest/plugins/datadog.md:108]: I think we are following this structure without tables in some other places as well. So, we can keep it as it is now.

**Comment by navendu-pottekkat** [comment]
> IIRC we agreed upon using `bash` as code highlighting syntax instead of `shell`. Please see [#7030 (comment)](https://github.com/apache/apisix/pull/7030#discussion_r871813072). Rest seems good to me

We are using "shell" in a lot of places and it seems to be working without issues right now. So, it was decided to keep using shell.

**Comment by navendu-pottekkat** [comment]
Ready for another round of review.

**Review by yzeng25 — APPROVED** [review]

**Pull request merged**